From 2f261905ca3d8c5c62c7d7173905cc570d995457 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Tue, 17 Sep 2024 14:55:26 +0200
Subject: [PATCH] gst-decoder.c: switch to decodebin3

decodebin3 is now prefered to decodebin till GStreamer-1.24 as explained here:
https://gstreamer.freedesktop.org/releases/1.24

This solves negotiation issues encountered with V4L2 stateless decoders
when using kmscube video playback option.

Change-Id: I4a2f836a16690b1cb765c7ac28b7a25a89adc68d
Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Upstream-Status: Submitted
---
 gst-decoder.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst-decoder.c b/gst-decoder.c
index 5431014..91bfe93 100644
--- a/gst-decoder.c
+++ b/gst-decoder.c
@@ -271,7 +271,7 @@ video_init(const struct egl *egl, const struct gbm *gbm, const char *filename)
 
 	/* Setup pipeline: */
 	static const char *pipeline =
-		"filesrc name=\"src\" ! decodebin name=\"decode\" ! video/x-raw ! appsink sync=false name=\"sink\"";
+		"filesrc name=\"src\" ! decodebin3 name=\"decode\" ! video/x-raw ! appsink sync=false name=\"sink\"";
 	dec->pipeline = gst_parse_launch(pipeline, NULL);
 
 	dec->sink = gst_bin_get_by_name(GST_BIN(dec->pipeline), "sink");
-- 
2.25.1

